src/index.js

```diff
@@ -19,30 +19,48 @@ var setup = state.current();
 
 // get into dom asap to trigger asset loading
 function renderHolder() {
-    var url = util.resolveClientUrl(util.currentRequestId(), true);
-    var html = '<div class="glimpse"><a class="glimpse-icon" target="_blank" href="' + url + '"><div class="glimpse-icon-text">Glimpse</div></a><div class="glimpse-hud"></div></div>'
-    $(html).appendTo('body');
+    if (document.readyState === 'complete') {
```

Comment:
maybe?  I'm seeing that on chrome & safari, when this was called the dom was in "interactive" state, but things seemed to work.  Only in firefox was this getting called when the dom was in "loading" state, which is where we saw the problem.